src/test/regress/sql/gp_query_id.sql

```diff
@@ -0,0 +1,159 @@
+--
+-- Test the query command identification
+--
+create extension if not exists gp_inject_fault;
+
+-- start_ignore
+drop function if exists sirv_function();
+drop function if exists not_inlineable_sql_func(i int);
+drop table if exists test_data1;
+drop table if exists test_data2;
+drop table if exists t;
+drop table if exists t1;
+drop table if exists t2;
+-- end_ignore
+
+set client_min_messages = notice;
+select gp_inject_fault('all', 'reset', dbid) from gp_segment_configuration;
+
+create or replace function sirv_function() returns text as $$
+declare
+    result1 text;
+    result2 text;
+    result3 text;
+begin
+    create table test_data1 (x int, y int) with (appendonly=true) distributed by (x);
+    create table test_data2 (x int, y varchar) with (appendonly=true) distributed by(x);
+
+    insert into test_data1 values (1,1);
+    insert into test_data1 values (1,2);
+
+    insert into test_data2 values (1, 'one');
+    insert into test_data2 values (1, 'ONE');
+
+    select case when count(*)>0 then 'PASS' else 'FAIL' end into result1 from test_data1;
+    select case when count(*)>0 then 'PASS' else 'FAIL' end into result2 from test_data2;
+
+    drop table test_data1;
+    drop table test_data2;
+
+    if (result1 = 'PASS')  and  (result2 = 'PASS') then
+        result3 = 'PASS';
+    else
+        result3 = 'FAIL';
+    end if;
+    return result3;
+end $$ language plpgsql volatile;
+
+\c
+
+select gp_inject_fault_infinite('track_query_command_id', 'skip', dbid) from gp_segment_configuration
+where role = 'p' and content = -1;
+
+select sirv_function();
+
+-- Test that the query command id is correct after execution of queries in the InitPlan
+create table t as select (select sirv_function()) as res distributed by (res);
+
+-- Test a simple query
+select * from t;
+
+drop table t;
+
+-- Test a cursor
+begin;
+declare cur1 cursor for select sirv_function() as res;
+fetch 1 from cur1;
+fetch all from cur1;
```

Comment:
What is the point of this command if the function returns only one row, which was already obtained above?